fe/fe-core/src/main/java/org/apache/doris/nereids/stats/HashJoinEstimation.java

```diff
@@ -0,0 +1,155 @@
+                return -1;
+            }
+            rowCount = rightStats.getRowCount();
+        } else {
+            if (leftStats.getRowCount() == -1) {
+                return -1;
+            }
+            rowCount = leftStats.getRowCount();
+        }
+        Map<Slot, ColumnStats> leftSlotToColStats = leftStats.getSlotRefToColumnStatsMap();
+        Map<Slot, ColumnStats> rightSlotToColStats = rightStats.getSlotRefToColumnStatsMap();
+        double minSelectivity = 1.0;
+        for (Expression eqJoinPredicate : eqConjunctList) {
+            long lhsNdv = leftSlotToColStats.get(eqJoinPredicate.child(0)).getNdv();
+            lhsNdv = Math.min(lhsNdv, leftStats.getRowCount());
+            long rhsNdv = rightSlotToColStats.get(eqJoinPredicate.child(1)).getNdv();
+            rhsNdv = Math.min(rhsNdv, rightStats.getRowCount());
+            // Skip conjuncts with unknown NDV on either side.
+            if (lhsNdv == -1 || rhsNdv == -1) {
+                continue;
+            }
+            // TODO: Do we need NULL_AWARE_LEFT_ANTI_JOIN type as stale optimizer?
+            double selectivity = 1.0;
+            switch (joinType) {
+                case LEFT_SEMI_JOIN: {
+                    selectivity = (double) Math.min(lhsNdv, rhsNdv) / (double) (lhsNdv);
+                    break;
+                }
+                case RIGHT_SEMI_JOIN: {
+                    selectivity = (double) Math.min(lhsNdv, rhsNdv) / (double) (rhsNdv);
+                    break;
+                }
+                case LEFT_ANTI_JOIN:
+                    selectivity = (double) (lhsNdv > rhsNdv ? (lhsNdv - rhsNdv) : lhsNdv) / (double) lhsNdv;
+                    break;
+                case RIGHT_ANTI_JOIN: {
+                    selectivity = (double) (rhsNdv > lhsNdv ? (rhsNdv - lhsNdv) : rhsNdv) / (double) rhsNdv;
+                    break;
+                }
+                default:
+                    throw new RuntimeException("joinType is not supported");
+            }
+            minSelectivity = Math.min(minSelectivity, selectivity);
+        }
+        Preconditions.checkState(rowCount != -1);
+        return Math.round(rowCount * minSelectivity);
+    }
+
+    private static long getJoinRowCount(StatsDeriveResult leftStats, StatsDeriveResult rightStats,
+            List<Expression> eqConjunctList, JoinType joinType) {
+        long lhsCard = leftStats.getRowCount();
+        long rhsCard = rightStats.getRowCount();
+        Map<Slot, ColumnStats> leftSlotToColumnStats = leftStats.getSlotRefToColumnStatsMap();
+        Map<Slot, ColumnStats> rightSlotToColumnStats = rightStats.getSlotRefToColumnStatsMap();
+        if (lhsCard == -1 || rhsCard == -1) {
+            return lhsCard;
+        }
+        long result = -1;
+        for (Expression eqJoinConjunct : eqConjunctList) {
+            Expression left = eqJoinConjunct.child(0);
+            if (!(left instanceof SlotReference)) {
```

Comment:
What about input conjunct child is not `SlotReference `? e.g., `a+1=b+2`

Reply:
Will support such case in the future, I guess there doesn't have such join condition in tpch